src/video/x11/SDL_x11video.c

```diff
@@ -307,6 +307,7 @@ X11_CreateDevice(void)
     device->StartTextInput = X11_StartTextInput;
     device->StopTextInput = X11_StopTextInput;
     device->SetTextInputRect = X11_SetTextInputRect;
+    device->HasScreenKeyboardSupport = X11_HasScreenKeyboardSupport;
```

Comment:
You should also implement and set `IsScreenKeyboardShown`, `ShowScreenKeyboard` and `HideScreenKeyboard` instead of adding the functionality to X11_StartTextInput()

Reply:
There are applications that rely on those APIs separately from Start/Stop text input.